### ISSUE TITLE
Show comma and period separators when inputting amount?

### ISSUE THREAD

**Issue by muinmomin** [BlueWallet/BlueWallet#4973]
When inputting amounts, especially in sats, it's hard to differentiate between 2000000 (2 million) vs 200000 (200k). Thoughts on adding commas (or periods based on locale) automatically to the input?

Example: user types in `2000`, input shows as either `2,000` (for en-US) or `2.000` (for pt-BR).

In some places, like the input screen where input could be error prone, the separators are not there, but in other places like the transaction list, they are:

| No separators  | Separators |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/8409428/185806369-4a7917eb-6136-4042-bd91-91c4e7717edd.PNG" width="250">  | <img src="https://user-images.githubusercontent.com/8409428/185806416-afda1205-0356-44ec-92a3-cd304d4619a1.PNG" width="250">  |

--

Happy to help with a PR if this sounds okay!

--

**Comment by Overtorment** [comment]
thats not easily doable because that place where you type is a `TextInput` with its own representation and behaviour. we cant change much without reimplementing whole thing from scratch.

correct me if im wrong @marcosrdz

**Comment by marcosrdz** [comment]
> thats not easily doable because that place where you type is a `TextInput` with its own representation and behaviour. we cant change much without reimplementing whole thing from scratch.
> 
> 
> 
> correct me if im wrong @marcosrdz 

Thats correct. The current implementation would require a complete write up

**Comment by muinmomin** [comment]
@Overtorment @marcosrdz Hm, unless BlueWallet has a different implementation of `TextInput`, I don't think that will be a limiting factor? Example: https://snack.expo.dev/B2v4TQRIR

<img width="324" alt="image" src="https://user-images.githubusercontent.com/8409428/185813125-f652ede4-a7f4-487e-b28b-84d830fb9740.png">

Android seems to have some [quirks](https://github.com/facebook/react-native/issues/19410) with `toLocaleString`, and there are definitely some things to think through for the BTC unit (which can have both decimal and thousands separators together, so we can't just strip them), but I think `TextInput` shouldn't an issue

**Comment by marcosrdz** [comment]
The current implementation is completely en-US focused. It needs to be rewritten to support all locales
